### PR TITLE
Bugfix create subFolders

### PR DESCRIPTION
**Pull request by JoGir**

### CODE REVIEW

File: qabelbox/src/androidTest/java/de/qabel/qabelbox/ui/files/FilesFragmentUITest.java [hunk 1]
```diff
@@ -100,38 +106,15 @@ public String getName() {
...
     }
 
-    public FilesFragmentUITest() throws Exception {
-        //setup data before MainActivity launched. This avoid the call to create identity
-        if (mFillAccount) {
+    public FilesFragmentUITest(){
+        try {
             setupData();
+        } catch (Exception e) {
+            Log.e(TAG, "Cannot setupData", e);
```
Comment (audax) [inline qabelbox/src/androidTest/java/de/qabel/qabelbox/ui/files/FilesFragmentUITest.java]: Pleae don't put setup code in the constructor. If you need to run setup code before the activity is launched, just set the IntentsTestRule to not launch automatically (last argument to 'false'), use proper @beforeClass methods and launch the activity in the setUp method with testRule.launchActvity(null);

File: qabelbox/src/androidTest/java/de/qabel/qabelbox/ui/files/FilesFragmentUITest.java [hunk 2]
```diff
@@ -179,12 +172,13 @@ public void shareFileTest() {
         //Check Contact is Visible
         onView(withText(testContact.getAlias())).check(matches(isDisplayed()));
 
+        //Perform share
         onView(withText(R.string.ok)).perform(click());
 
-        UITestHelper.sleep(1000);
+        UITestHelper.sleep(200);
```
Comment (audax) [inline qabelbox/src/androidTest/java/de/qabel/qabelbox/ui/files/FilesFragmentUITest.java]: No sleeps in UI test code, Espresso has implicit waits built in

File: qabelbox/src/androidTest/java/de/qabel/qabelbox/ui/files/FilesFragmentUITest.java [hunk 3]
```diff
@@ -214,18 +207,58 @@ public void shareFileTest() {
...
+        onView(withId(R.id.fab)).check(matches(isDisplayed())).perform(click());
+        onView(withText(R.string.create_folder)).check(matches(isDisplayed())).perform(click());
+
+        onView(allOf(withClassName(endsWith("EditTextFont")))).perform(typeText(CREATE_FOLDER_TEST_NAME));
+        onView(withText(R.string.ok)).perform(click());
+    }
+
+    @Test
+    public void testCreateFolder() {
+        goToFiles();
```
Comment (audax) [inline qabelbox/src/androidTest/java/de/qabel/qabelbox/ui/files/FilesFragmentUITest.java]: Testing the drawer is not useful. You can add an Intent-extra to the launch Intent of the MainActivity which brings you directly to the files fragment. (It doesn't exist yet ;) )

File: qabelbox/src/androidTest/java/de/qabel/qabelbox/ui/CreateBoxAccountUITest.java [hunk 4]
```diff
@@ -73,13 +69,24 @@
 public class CreateBoxAccountUITest extends UIBoxHelper {
 
     @Rule
-    public ActivityTestRule<CreateAccountActivity> mActivityTestRule = new ActivityTestRule<>(CreateAccountActivity.class, false, true);
+    public ActivityTestRule<CreateAccountActivity> mActivityTestRule = new ActivityTestRule<CreateAccountActivity>(CreateAccountActivity.class, false, true){
```
Comment (audax) [inline qabelbox/src/androidTest/java/de/qabel/qabelbox/ui/CreateBoxAccountUITest.java]: This works, but I like it better to just not launch the activity automatically but with testRule.launchActivity.

File: qabelbox/src/androidTest/java/de/qabel/qabelbox/ui/files/FilesFragmentUITest.java [hunk 5]
```diff
@@ -156,10 +158,10 @@ public void setUp() throws IOException, QblStorageException {
         wakeLock = UIActionHelper.wakeupDevice(mActivity);
         mSystemAnimations = new SystemAnimations(mActivity);
         mSystemAnimations.disableAll();
+        mActivity.runOnUiThread(() -> mActivity.selectIdentity(testIdentity));
```
Comment (audax) [inline qabelbox/src/androidTest/java/de/qabel/qabelbox/ui/files/FilesFragmentUITest.java]: if you don't run the activity automatically, you don't need to use such a hack. Just select the activity service and then start the activity.

File: qabelbox/src/androidTest/java/de/qabel/qabelbox/ui/files/FilesFragmentUITest.java [hunk 6]
```diff
@@ -134,14 +123,27 @@ private void setupData() throws IOException, QblStorageException {
...
 
+    @AfterClass
+    public static void cleanUpData() throws QblStorageException {
+        mBoxHelper.deleteFolder(CREATE_FOLDER_TEST_NAME, testIdentity, "");
+        mBoxHelper.deleteFolder(TEST_FOLDER, testIdentity, "");
+        for (ExampleFile exampleFile : exampleFiles) {
+            mBoxHelper.deleteFile(InstrumentationRegistry.getContext(), testIdentity, exampleFile.getName(), "");
+        }
+        mBoxHelper.deleteIdentity(testIdentity);
+        mBoxHelper.deleteIdentity(testIdentity2);
```
Comment (audax) [inline qabelbox/src/androidTest/java/de/qabel/qabelbox/ui/files/FilesFragmentUITest.java]: There is no need to delete those files. Just remove the identities. Other tests can't depend on a clean environment anyway.

**Comment by jmt85** [comment]
please rebase/merge

**Pull request merged**